pkg/istio-agent/agent.go

```diff
@@ -205,6 +210,35 @@ func NewAgent(proxyConfig *mesh.ProxyConfig, agentOpts *AgentOptions, sopts *sec
 		cfg:         agentOpts,
 		secOpts:     sopts,
 		envoyOpts:   eopts,
+		watcher:     filewatcher.NewWatcher(),
+		tlsOpts:     &citadel.TLSOptions{},
+	}
+}
+
+func (a *Agent) watcherHandler(ctx context.Context) {
+	for {
+		select {
+		case gotEvent := <-a.watcher.Events(a.tlsOpts.RootCert):
+			log.Infof("Receive file %s event %v", a.tlsOpts.RootCert, gotEvent)
+			if a.caClient != nil {
+				err := a.caClient.InitClient()
+				if err != nil {
+					log.Warnf("Failed to init CA client")
+					continue
+				}
+
+				a.secretCache.CallUpdateCallback(security.RootCertReqResourceName)
+				a.secretCache.CallUpdateCallback(security.WorkloadKeyCertResourceName)
+
+				err = a.xdsProxy.InitIstiodDialOptions(a)
```

Comment:
same concern about data race with the citadel client, except this one is introduce in this PR